### PR TITLE
feat: update commander to v9

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
Updated snapshots and tests.

**If relevant, did you update the documentation?**
No

**Summary**

- Update commander to v9
- Commander v9 provides suggestions for unknown options by default. Removed our custom logic for suggestion options and kept the code for suggesting commands.
- Ideally, the commander should export its [API for suggestSimilar](https://github.com/tj/commander.js/blob/master/lib/suggestSimilar.js#L95)

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

File: test/build/unknown/__snapshots__/unknown.test.js.snap.webpack5 [hunk 1]
```diff
@@ -37,56 +37,55 @@ exports[`unknown behaviour should log an error if an unknown flag is passed and
...
+(Did you mean --entry?)
 [webpack-cli] Run 'webpack --help' to see available commands and options"
 `;
 
 exports[`unknown behaviour should log an error if an unknown flag is passed and suggests the closest match to an unknown flag using "b" command: stdout 1`] = `""`;
 
 exports[`unknown behaviour should log an error if an unknown flag is passed and suggests the closest match to an unknown flag using "bundle" command: stderr 1`] = `
 "[webpack-cli] Error: Unknown option '--entyr'
-[webpack-cli] Did you mean '--entry'?
+(Did you mean --entry?)
```
Comment (alexander-akait) [inline test/build/unknown/__snapshots__/unknown.test.js.snap.webpack5:64]: Will be great to have `[webpack-cli]` prefix here, is there the places where we can hook?

**Review by alexander-akait — COMMENTED** [review]
Looks good, one note

File: test/build/unknown/__snapshots__/unknown.test.js.snap.webpack5 (hunk 1 above)
Comment (snitin315) [reply test/build/unknown/__snapshots__/unknown.test.js.snap.webpack5]: Yes, we can disable it with `this.program.showSuggestionAfterError(false);` and keep our own logic.

**Comment by snitin315** [comment]
@alexander-akait Done 👍🏻

**Review by alexander-akait — APPROVED** [review]
Feel free to merge :star:

**Pull request merged**